File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> freq(27, 0);

        for (auto c : s)
            freq[c - 'a']++;
            
        for (auto i = 0; i < s.size(); i++)
            if (freq[s[i] - 'a'] == 1)
                return i;
        return -1;
    }
};

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.go
func firstUniqChar(s string) int {
    var letters [26]int

    for _, v := range s {
        letters[v - 'a']++
    }

    for i, v := range s {
        if letters[v - 'a'] == 1 {
            return i
        }
    }

    return -1
}